docs/decisions/0005-architecture-and-data-modeling.rst

```diff
@@ -0,0 +1,149 @@
+- Services can request authorization decisions by calling the Enforcement API with the necessary context (e.g., user, action, resource) with optional context and receiving a decision (allow/deny) in response.
+
+#. Data & Storage Model
+========================
+
+Use a MySQL as the Main Backend for the Policy Store
+----------------------------------------------------
+- Use MySQL as the main backend for persistent storage for policies (policy store), leveraging Casbin's Django ORM adapter for integration and our own modifications to the adapter as needed in our own ``engine/``` package.
+- Use the same schema used by Casbin's Django ORM adapter, which includes a table for storing policies with:
+  - ``id``: A unique identifier for each policy. This depends on the database backend. In our case, MySQL uses an auto-incrementing integer.
+  - ``ptype``: The policy type (e.g., ``p`` for policy, ``g, g2`` for grouping).
+  - ``v0, v1, v2, v3, v4, v5``: The policy fields that define the subject, action, object, and context. The exact meaning of these fields depends on the policy type and model configuration.
+- Optionally, include additional metadata fields in the policy table to support auditing and versioning.
+
+Store All Policies in the Policy Store
+--------------------------------------
+- All policies (i.e., any type of rule) should be stored in the policy store to ensure a single source of truth for authorization.
+- Use the policy store to manage RBAC mappings, such as user-role and role-permission assignments, using Casbin's grouping policies (``g, g2``).
+- Use Casbin's adapter APIs based on Django APIs to load policies from the policy store into the Authorization Engine at startup and whenever policies are updated.
+
+Maintain Consistent Model and Policy Definitions Across Services
+----------------------------------------------------------------
+- Policies should be defined in a way that is consistent across services, using the same naming conventions and structures for subjects, actions, objects, and contexts. For example, if the LMS defines a policy for "viewing a course" the CMS should use the same terminology and structure when defining a similar policy for "viewing content".
+- Each column in the policy table (v0, v1, v2, etc.) should have a consistent meaning across services. For example, for the same type ``v0`` should always represent the subject, ``v1`` the action, ``v2`` the object, and so on.
+
+#. Client Interactions with the Authorization System
+=====================================================
+
+Use the Enforcement API for Authorization Decisions for External Clients
+------------------------------------------------------------------------
+- External clients (e.g., MFEs, IDAs, or any service not co-located with the policy store) must use the REST API provided by the Open edX authorization layer to request authorization decisions.
```

Comment:
Can you be specific about which service is going to host this? I assume it will have to be LMS, but I'd like it to be explicit. Things I've read elsewhere seem to indicate that Studio may have its own store? In which case would external services have to query both? Or would both sets of policies be stored in a single LMS-based store with a single set of REST endpoints?